codebase/src/sorting/components/ArrayInput.vue

```diff
@@ -1,258 +1,283 @@
-        <div class="form-container">
-            <form @submit.prevent="handleSubmit" class="form-styles">
-                <input v-model="inputValue" type="text" placeholder="Enter number" />
-                <button type="submit">Add</button>
-                <button type="button" @click="clearArray">Clear</button>
-                <button type="button" @click="sortArray">Sort</button>
-                <button type="button" @click="generateArray">Random</button>
-            </form>
-            <p v-if="error" class="error">{{ error }}</p>
-        </div>
-        <div class="visualization-container">
-            <div class="numbers-container">
-                <div 
-                    v-for="(number, index) in array" 
-                    :key="index" 
-                    class="number-div"
-                    :class="{
-                        'left-index': index === currentLeftIndex,
-                        'right-index': index === currentRightIndex,
-                        'sorted-index': currentSortedIndex !== null && index >= currentSortedIndex
-                    }"
-                    :style="{ 
-                        height: `${number * (number / Math.max(...array)) * 7}px`,
-                        width: `${array.length < 20 ? 20 : 5}px`
-                    }"
-                >
-                    <span v-if="array.length <= 15">{{ number }}</span>
-                </div>
-            </div>
-        </div>
-        <div class="pseudo-code-container">
-            <div 
-                v-for="(line, index) in pseudoCode" 
-                :key="index" 
-                :class="{ 'highlighted': currentLines.includes(index) }"
-                :style="{ paddingLeft: `${(line.match(/^(\s*)/)?.[0].length || 0) * 2}vw` }"
-            >
-                {{ line.trim() }}
-            </div>
+  <div class="arrayInput">
+    <p class="array-update">Array so far: {{ array }}</p>
+    <div class="form-container">
+      <form @submit.prevent="handleSubmit" class="form-styles">
+        <input v-model="inputValue" type="text" placeholder="Enter number" />
+        <button type="submit">Add</button>
+        <button type="button" @click="clearArray">Clear</button>
+        <button type="button" @click="sortArray">Sort</button>
+        <button type="button" @click="generateArray">Random</button>
+      </form>
+      <p v-if="error" class="error">{{ error }}</p>
+    </div>
+    <div class="visualization-container">
+      <div class="numbers-container">
+        <div
+          v-for="(number, index) in array"
+          :key="index"
+          class="number-div"
+          :class="{
+            'left-index': index === currentLeftIndex,
+            'right-index': index === currentRightIndex,
+            'sorted-index':
+              currentSortedIndex !== null && index >= currentSortedIndex,
+          }"
+          :style="{
+            height: `${number * (number / Math.max(...array)) * 7}px`,
```

Comment:
Avoid doing heavy logic in the template. I would move it out to several computed properties that are named descriptively, then put those computed refs in the template to make it more declarative